### PR TITLE
added prot enabled

### PR DESCRIPTION
**Pull request by Hobbydos**
Proposal for [issue 702](https://github.com/openconfig/featureprofiles/issues/702)

### CODE REVIEW

**Comment by Hobbydos** [comment]
cc @robshakir  @xw-g

**Comment by xw-g** [comment]
Because current OC schema does not specify the default value, also because we are doing the same thing for other protocols (like [static protocl](https://github.com/openconfig/featureprofiles/blob/06a95833b93b02dbfbe7dfc918d8124ceba89620/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go#L203)), I LGTM this change. 

cc: @vcheny @vosipchu @sachendras thoughts?

**Comment by sachendras** [comment]
LGTM

**Comment by mojiiba** [comment]
@xw-g @Hobbydos  can we manage this via a deviation flag? I was going to open a pull request to do the reverse of this pR was doing.

**Comment by mojiiba** [comment]
Also, I noticed several tests configure static routes and there are slight differences between  them. Can we have a util function that unify this? I can open a pull request if you want.

**Comment by mojiiba** [comment]
please ignore my latest comment

**Comment by Hobbydos** [comment]
> @Hobbydos can you also help update related part in OTG? Thanks.

I'll work on doing OTG - Will just run a few tests for now.



> @xw-g @Hobbydos can we manage this via a deviation flag? I was going to open a pull request to do the reverse of what this pR is doing.

Is there any purpose for this being a deviation?

**Comment by vcheny** [comment]
Does it make sense to make enabled leafs in the OC model as true by default? Would imagine enable leaf to be True when adding a protocol most of time.

**Comment by mojiiba** [comment]
> Does it make sense to make enabled leafs in the OC model as true by default? Would imagine enable leaf to be True when adding a protocol most of time.

yes. I also think some of enabled leaves can be removed from the oc model since the intentions for the enabled leaves are not clear. One example is vrf/enabled that is not clear what device should do with it, especially when they ate set to false. There is no way for most vendors to define a vrf and the disable it. We have to note that the devices still have to provide CLI. Making OC model cli friendl very helpful to adaption of the oc.

**Comment by mojiiba** [comment]
> > @Hobbydos can you also help update related part in OTG? Thanks.
> 
> I'll work on doing OTG - Will just run a few tests for now.
> 
> > @xw-g @Hobbydos can we manage this via a deviation flag? I was going to open a pull request to do the reverse of what this pR is doing.
> 
> Is there any purpose for this being a deviation?

setting enabled will make our device to fail the test. managing this via deviation will enable both of us to pass the test for now. Google team can decide to the desired behavior and we can dap

> > @Hobbydos can you also help update related part in OTG? Thanks.
> 
> I'll work on doing OTG - Will just run a few tests for now.
> 
> > @xw-g @Hobbydos can we manage this via a deviation flag? I was going to open a pull request to do the reverse of what this pR is doing.
> 
> Is there any purpose for this being a deviation?

this change will make our device fail. The flag will allow both of us to pass the tests for now. @xw-g and their team can decide on the desired behavior and we can comply accordingly.

**Comment by xw-g** [comment]
@mojiiba which logic you are planning to reverse? [this one](https://github.com/openconfig/featureprofiles/blob/06a95833b93b02dbfbe7dfc918d8124ceba89620/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go#L203)?

To keep things moving, shall we do the following? 
1. Let this PR move forward and merge, because current OC schema does not set the value by default.
2. @xw-g to file an OC PR to set the `enabled` as true by default (might take some time through the OC community, or get rejected).
3. @mojiiba to work on a PR to introduce a deviation flag for `enabled`? 

thoughts? @mojiiba, @vosipchu?

**Comment by mojiiba** [comment]
> @mojiiba which logic you are planning to reverse? [this one](https://github.com/openconfig/featureprofiles/blob/06a95833b93b02dbfbe7dfc918d8124ceba89620/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go#L203)?
> 
> To keep things moving, shall we do the following?
> 
>     1. Let this PR move forward and merge, because current OC schema does not set the value by default.
> 
>     2. @xw-g to file an OC PR to set the `enabled` as true by default (might take some time through the OC community, or get rejected).
> 
>     3. @mojiiba to work on a PR to introduce a deviation flag for `enabled`?
> 
> 
> thoughts? @mojiiba, @vcheny, @vosipchu, @sachendras ?

sounds good, thank you!

File: feature/experimental/isis/ate_tests/internal/session/session.go [hunk 1]
```diff
@@ -105,6 +105,7 @@ func ProtocolPath() *networkinstance.NetworkInstance_ProtocolPath {
 func addISISOC(dev *oc.Root, areaAddress, sysID, ifaceName string) {
 	inst := dev.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance)
 	prot := inst.GetOrCreateProtocol(PTISIS, ISISName)
+	prot.Enabled = ygot.Bool(true)
```
Comment (aredmon8551) [inline feature/experimental/isis/ate_tests/internal/session/session.go:108]: Curious - why add this?  I don't think enabling/disabling individual routing protocols is widely supported.

Comment (aredmon8551) [reply feature/experimental/isis/ate_tests/internal/session/session.go:108]: Looks like I'm wrong - IOS-like NOS's seem to support a 'shutdown' command under protocols.  Still curious about the utility of this leaf for this test.

Comment (xw-g) [reply feature/experimental/isis/ate_tests/internal/session/session.go:108]: @aredmon8551 does this [comment](https://github.com/openconfig/featureprofiles/pull/745#issuecomment-1314620480) answer your question? We are not enforcing it. We want to keep things moving while seeking clarification at the source (the OC)

Comment (aredmon8551) [reply feature/experimental/isis/ate_tests/internal/session/session.go:108]: Thanks!

Somewhat answers the question - though I'm missing the motivation for the addition - is this change (plus the addition of a deviation) needed to allow support for all NOS's?  Or is this for consistency?  I'm not aware of this being a mandatory leaf/config node in any implementation.  Perhaps I'm missing something.

**Comment by robshakir** [comment]
> @mojiiba which logic you are planning to reverse? [this one](https://github.com/openconfig/featureprofiles/blob/06a95833b93b02dbfbe7dfc918d8124ceba89620/feature/gribi/ate_tests/ack_in_presence_other_routes/route_ack_test.go#L203)?
> 
> To keep things moving, shall we do the following?
> 
> 1. Let this PR move forward and merge, because current OC schema does not set the value by default.
> 2. @xw-g to file an OC PR to set the `enabled` as true by default (might take some time through the OC community, or get rejected).
> 3. @mojiiba to work on a PR to introduce a deviation flag for `enabled`?
> 
> thoughts? @mojiiba, @vcheny, @vosipchu, @sachendras ?

SGTM too.

File: feature/experimental/isis/ate_tests/internal/session/session.go (hunk 1 above)
Comment (xw-g) [reply feature/experimental/isis/ate_tests/internal/session/session.go:108]: The referenced comment basically suggests to push things forward via 2 tracks in parallel: 

1. Align with OC. The OC doesn't set default value, so we need to set the value. That's what this PR is about. Meanwhile, let's create a deviation for other vendors that does not support set it explicitly. 
3. I'm going to file a PR to OC suggesting set the default value true. 
4. If 2 merged, let's converge on what test behaviors we want. In that case, probably we will remove the requirement of explicitly set it true.

Comment (aredmon8551) [reply feature/experimental/isis/ate_tests/internal/session/session.go:108]: Ack, makes sense.  Thanks!

**Comment by xw-g** [comment]
@Hobbydos ping me again when you have OTG updated. Thanks.

**Comment by Hobbydos** [comment]
@xw-g added commit for OTG change

**Review by xw-g — APPROVED** [review]

**Pull request merged**